lib/chef/knife/vsphere_vm_network_add.rb

```diff
@@ -10,8 +10,19 @@
 class Chef::Knife::VsphereVmNetworkAdd < Chef::Knife::BaseVsphereCommand
   banner 'knife vsphere vm network add VMNAME NETWORKNAME'
 
+  option :adapter_type,
+         long: '--adapter-type STRING',
+         description: 'Adapter type eg e1000,vmxnet3',
+         required: false
```

Comment:
If the user doesn't supply something they'll get a fatal exit later down the road. I think we either need a default value or to mark this as required.